src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -116,14 +118,11 @@ private static void setupDrivers(Application application) {
         DriverFeature.updateDriverInfo();
     }
     private static void setupDriverCountTest(Application application, DriverManager driverManager) {
-        application.addTest("Count Drivers", (ActionEvent e) -> {
-            DriverCounterVisitor counter = new DriverCounterVisitor();
-            for (Job2dDriver driver : driverManager.getAllDrivers()) {
-                if (driver instanceof VisitableDriver) {
-                    ((VisitableDriver) driver).accept(counter);
-                }
-            }
-            logger.info(String.valueOf(counter.getCount()));;
+        DriverCounterService counterService = new DriverCounterService(driverManager);
+
+        application.addTest("Count Drivers", (ActionEvent s) -> {
+            Map<String, Integer> counts = counterService.getCounts();
+            counts.forEach((type, count) -> logger.info(type + " count: " + count));
```

Comment:
The code of this test can also be moved to DriverCounterService.java

Reply:
done